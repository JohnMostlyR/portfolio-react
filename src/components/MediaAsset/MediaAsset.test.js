import React from 'react';
import {shallow} from 'enzyme';

import MediaAsset, {Figure, ImageWrapper} from './index';

describe('MediaAsset', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MediaAsset/>);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a Figure component', () => {
    expect(wrapper.find(Figure)).toHaveLength(1);
  });

  it('that should receive an idOdd prop', () => {
    expect(wrapper.find(Figure).prop('isOdd')).toBeFalsy();
    wrapper.setProps({ isOdd: true, });
    expect(wrapper.find(Figure).prop('isOdd')).toBe(true);
  });

  it('should have a ImageWrapper component', () => {
    expect(wrapper.find(ImageWrapper)).toHaveLength(1);
  });

  it('that should receive an imageSource prop', () => {
    const myImage = `${Date.now()}.png`;
    wrapper.setProps({ imageSource: myImage, });
    expect(wrapper.find(ImageWrapper).prop('imageSource')).toBe(myImage);
  });
});