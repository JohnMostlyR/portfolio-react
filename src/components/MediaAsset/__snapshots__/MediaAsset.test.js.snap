// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaAsset should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaAsset />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <Styled(ResponsiveImage)
                imageAlt="Project"
                imageSource={undefined}
                ratio="16by9"
        />
</styled.div>,
      "isOdd": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(ResponsiveImage)
          imageAlt="Project"
          imageSource={undefined}
          ratio="16by9"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "imageAlt": "Project",
          "imageSource": undefined,
          "ratio": "16by9",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <Styled(ResponsiveImage)
                    imageAlt="Project"
                    imageSource={undefined}
                    ratio="16by9"
          />
</styled.div>,
        "isOdd": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(ResponsiveImage)
            imageAlt="Project"
            imageSource={undefined}
            ratio="16by9"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "imageAlt": "Project",
            "imageSource": undefined,
            "ratio": "16by9",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
